### PR TITLE
WIP Fixes #2049 Silence stack traces in test output

### LINKED ISSUE #2049: Silence expected stack traces in test output
1. Steal [silenceSystemErr](https://github.com/typelevel/cats-effect/blob/c7af84bc2d3ae63d77336d521818cb05c66579cf/core/shared/src/test/scala/cats/effect/internals/TestUtils.scala#L31), with credit
2. Find the tests that are dumping stack traces
3. Wrap them in `silenceSystemErr`
4. Enjoy cleaner Travis runs

### PR DESCRIPTION
**Pull request by ashwinbhaskar**

### CODE REVIEW

File: testing/src/main/scala/org/http4s/testing/ErrorReportingUtils.scala [hunk 1]
```diff
@@ -12,7 +12,7 @@ import java.io.{ByteArrayOutputStream, PrintStream}
 
 import scala.util.control.NonFatal
 
-@deprecated("Not related to HTTP, and will be removed from public interface.", "0.21.0-RC2")
+//@deprecated("Not related to HTTP, and will be removed from public interface.", "0.21.0-RC2")
```
Comment (rossabaker) [inline testing/src/main/scala/org/http4s/testing/ErrorReportingUtils.scala:15]: There is a copy in `ErrorReporting` that's in the test sources, but available internally.  If you use that, we can leave this one deprecated.  This one is in main, and it's not an API that we wanted to publicly support in an HTTP library.

File: testing/src/test/scala/org/http4s/testing/ErrorReporting.scala [hunk 2]
```diff
@@ -11,9 +11,33 @@ package org.http4s.testing
...
+import cats.implicits._
+import cats.Monad
+import org.http4s.{MessageFailure, Request, Response, Status, Headers}
+import org.http4s.headers.{Connection, `Content-Length`}
+import org.http4s.syntax.string._
 
 object ErrorReporting {
 
+  /**
+   * Returns an ErrorHandler that does not Silence logging
```
Comment (rossabaker) [inline testing/src/test/scala/org/http4s/testing/ErrorReporting.scala:23]: This comment contradicts what the method does.

File: testing/src/test/scala/org/http4s/testing/ErrorReporting.scala [hunk 3]
```diff
@@ -31,11 +55,11 @@ object ErrorReporting {
...
         case NonFatal(e) =>
-          System.setErr(oldErr)
+//          System.setErr(oldErr)
           // In case of errors, print whatever was caught
-          fakeErr.close()
-          val out = outStream.toString("utf-8")
-          if (out.nonEmpty) oldErr.println(out)
+//          fakeErr.close()
+//          val out = outStream.toString("utf-8")
+//          if (out.nonEmpty) oldErr.println(out)
```
Comment (rossabaker) [inline testing/src/test/scala/org/http4s/testing/ErrorReporting.scala:62]: We should leave this method as it was, as it's still useful for silencing things actually logged to `System.err`.

**Comment by rossabaker** [comment]
Nothing we're doing here is binary incompatible, so we could cherry-pick this onto 0.21 when it's ready and minimize the divergence between the branches.

**Comment by ashwinbhaskar** [comment]
closing this pull request in favor of https://github.com/http4s/http4s/pull/3456, which is raised against series/0.21

**Pull request closed without merging**